lib/propolis/src/accessors.rs

```diff
@@ -10,189 +19,458 @@ use crate::vmm::{MemCtx, VmmHdl};
 
 type AccId = usize;
 
+const ID_NULL: AccId = 0;
 const ID_ROOT: AccId = 1;
 
-type TreeBackref<T> = Arc<Mutex<Tree<T>>>;
-
+struct TreeNode<T> {
+    parent_id: AccId,
+    node_ref: Weak<Node<T>>,
+    children: Option<BTreeSet<AccId>>,
```

Comment:
Is the `Option` here just to be able to `take` the children below? `std::mem::take` would also work without the Option.

Reply:
More to just avoid allocating the Set at all if there are no children.

Reply:
I can change it if you think that's needless optimization.

Reply:
ah, `BTreeSet::new` doesn't allocate so should be fine without the Option.

Reply:
Done